src/new/wallet.js

```diff
@@ -533,7 +715,96 @@ class HathorWallet extends EventEmitter {
     };
   }
 
+  /**
+   * Get balance for a token (same as old method to be used for compatibility)
+   *
+   * @params {string} tokenUid Token uid
+   *
+   * @return {Object} Object with balance { available, locked }
+   *
+   * @memberof HathorWallet
+   * @inner
+   **/
+  getOldBalance(tokenUid) {
+    storage.setStore(this.store);
+    const uid = tokenUid || this.token.uid;
+    const historyTransactions = this.getOldHistory();
+    return wallet.calculateBalance(Object.values(historyTransactions), uid);
+  }
+
+  /**
+   * Get full wallet history (same as old method to be used for compatibility)
+   *
+   * @return {Object} Object with transaction data { tx_id: { full_transaction_data }}
+   *
+   * @memberof HathorWallet
+   * @inner
+   **/
+  getOldHistory() {
+    storage.setStore(this.store);
+    const data = wallet.getWalletData();
+    const history = 'historyTransactions' in data ? data['historyTransactions'] : {};
+    return history;
+  }
+
+  /**
+   * Prepare history and balance and save on storage
+   * to prepare for the new signatures
+   *
+   * @memberof HathorWallet
+   * @inner
+   **/
+  prepareHistoryAndBalanceByToken() {
+    storage.setStore(this.store);
+    const history = this.getOldHistory();
+    const tokensHistory = {};
+    // iterate through all txs received and map all tokens this wallet has, with
+    // its history and balance
+    for (const tx of Object.values(history)) {
+      // we first get all tokens present in this tx (that belong to the user) and
+      // the corresponding balances
+      const balances = this.getTxBalance(tx);
+      for (const [tokenUid, tokenTxBalance] of Object.entries(balances)) {
+        let tokenHistory = tokensHistory[tokenUid];
+        if (tokenHistory === undefined) {
+          tokenHistory = [];
+          tokensHistory[tokenUid] = tokenHistory;
+        }
+        // add this tx to the history of the corresponding token
+        tokenHistory.push({
+          txId: tx.tx_id,
+          timestamp: tx.timestamp,
+          tokenUid,
+          balance: tokenTxBalance,
+          isVoided: tx.is_voided,
+        });
+      }
+    }
+
+    const tokensBalance = {};
+    for (const tokenUid of Object.keys(tokensHistory)) {
+      const totalBalance = this.getOldBalance(tokenUid);
+      // update token total balance
+      tokensBalance[tokenUid] = totalBalance;
+    }
+
+    // in the end, sort (in place) all tx lists in descending order by timestamp
+    for (const txList of Object.values(tokensHistory)) {
+      txList.sort((elem1, elem2) => elem2.timestamp - elem1.timestamp);
+    }
+
+    storage.setItem('old-facade:tokens', Object.keys(tokensHistory));
```

Comment:
I don't like this `old-facade` prefix. What does it mean? I don't think we are planning to remove this facade so I don't get why we are using the term "old" here as well.

Reply:
Well done! Great!